apps/sms/services/js/activity/client.js

```diff
@@ -0,0 +1,85 @@
+/* global EventDispatcher,
+          bridge
+*/
+
+(function(exports) {
+  'use strict';
+
+  /**
+   * Name of the service that is responsible for managing activities.
+   * @type {string}
+   */
+  const SERVICE_NAME = 'activity-service';
+
+  /**
+   * Reference to active bridge client instance.
+   * @type {Client}
+   */
+  var client = null;
+
+  /**
+   * Indicates if there is pending activity request.
+   * @type {boolean}
+   */
+  var hasPendingRequest = false;
+
+  var ActivityClient = {
+    /**
+     * Initialized activity service client bridge.
+     */
+    init() {
+      client = bridge.client(SERVICE_NAME);
+
+      client.on('activity-request', (request) => {
+        hasPendingRequest = true;
+        this.emit(request.name + '-activity-request', request.data);
+      });
+    },
+
+    /**
+     * Posts results to the activity caller.
+     * @param {*?} result
+     * @returns {Promise} Promise that resolves once operation is completed.
+     */
+    postResult(result) {
+      if (!hasPendingRequest) {
+        return Promise.reject(
+          new Error('There is no any activity request to post result to!')
+        );
+      }
+
+      hasPendingRequest = false;
+      return client.method('postResult', result);
+    },
+
+    /**
+     * Notifies activity caller that exception occurred.
+     * @param {*?} error
+     * @returns {Promise} Promise that resolves once operation is completed.
+     */
+    postError(error) {
+      if (!hasPendingRequest) {
+        return Promise.reject(
+          new Error('There is no any activity request to post error to!')
+        );
+      }
+
+      hasPendingRequest = false;
+      return client.method('postError', error);
+    },
+
+    /**
+     * Checks if client has pending activity request.
+     * @returns {boolean}
+     */
+    hasPendingRequest() {
```

Comment:
I need some sync (and for the case when service isn't created) way to determine if view is in activity mode